File: src/Tiantong.Wms.Api/Entities/Good.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Tiantong.Wms.Api
{
  [Table("goods")]
  public class Good : Entity
  {
    public virtual int warehouse_id { get; set; }

    public virtual string number { get; set; }

    public virtual string name { get; set; }

    public virtual string comment { get; set; }

    public virtual bool is_enabled { get; set; } = true;

    public virtual bool is_deleted { get; set; } = false;

    public virtual DateTime created_at { get; set; } = DateTime.Now;

    [ForeignKey("good_id")]
    public virtual List<Item> items { get; set; }

  }
}

File: src/Tiantong.Wms.Client/src/Entities/Good.ts
import Item from './Item'
import { DateTime } from '@/utils/common'

export default class Good {
  id: number = 0

  warehouse_id: number = 0

  number: string | null = null

  name: string = ''

  comment: string = ''

  is_enabled: boolean = true

  created_at: string = DateTime.now

  items: Array<Item>

  constructor(items: Item[] = []) {
    this.items = items
  }

}
